docs/core/diagnostics/diagnosticsource-instrumentation-consumption.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Instrument Code to Create DiagnosticSource Events and Consume Data with DiagnosticListener
+description: A tutorial for instrumenting code with DiagnosticSource and consuming data with DiagnosticListener
+ms.date: 05/02/2022
+---
+# Instrument Code to Create DiagnosticSource Events and Consume Data with DiagnosticListener
+
+**This article applies to: ✔️** .NET Core 3.1 and later versions **✔️** .NET Framework 4.5 and later versions
+
+This walkthrough shows how to instrument code with <xref:System.Diagnostics.DiagnosticSource?displayProperty=nameWithType>.
+It then moves on to explain how <xref:System.Diagnostics.DiagnosticSource?displayProperty=nameWithType> consumes the data by first
+finding the specified DiagnosticListeners one is interested in, subscribing to them, and decoding payloads for more specific data.
+It finishes by describing filtering, which controls which events pass through the system.
+
+-------------------------------------------
+
+## Instrumenting with DiagnosticSource/DiagnosticListener
+
+We show how to instrument DiagnosticListener using the DiagnosticListener we made in the [Getting Started Guide](/docs/core/diagnostics/diagnosticsource-getting-started).
+
+The `DiagnosticSource` logging
+interface consists of two methods:
+
+```C#
+    bool IsEnabled(string name)
+    void Write(string name, object value);
+```
+
+A typical call site will look like:
+
+```C#
+    if (httpLogger.IsEnabled("RequestStart"))
+        httpLogger.Write("RequestStart", new { Url="http://clr", Request=aRequest });
+```
+
+Architectural elements shown in the above code are as follows.
+Every event has a `string` name (e.g. `RequestStart`), and exactly one `object` as a payload.
+If you need to send more than one item, you can do so by creating an `object` with all information
+in it as properties. C#'s [anonymous type](https://msdn.microsoft.com/library/bb397696.aspx)
```

Comment:
```suggestion
in it as properties. C#'s [anonymous type](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/types/anonymous-types)
```
Or is there a linkage in-between the docs that suits better?
The MSDN link is outdated.